### ISSUE TITLE
Can Filex and USBX work on one storage medium at the same time？

### ISSUE THREAD

**Issue by arilink** [azure-rtos/usbx#32]
I have a question. I want to use an SD card or Flash as a storage medium, and then make a USB MSG device so that the PC can read and write the files in it. At the same time, STM32 uses FileX to access the files inside. Is this allowed?

**Comment by xiaocq2001** [comment]
No, it's not recommended. USB MSC device modifies sectors which makes conflicts with FileX.

**Comment by zhangshengping** [comment]
> No, it's not recommended. USB MSC device modifies sectors which makes conflicts with FileX.

is any plan to imply mtp class device?
@xiaocq2001

**Comment by xiaocq2001** [comment]
Please try PIMA class, with UX_PIMA_WITH_MTP_SUPPORT defined.

**Comment by arilink** [comment]
@xiaocq2001 If it is not operated at the same time, the system starts FileX to operate the SD card at the beginning of the system, and starts USB MSC to update the files on the SD card when needed. FileX only reads the files. Can this avoid conflicts?

**Comment by xiaocq2001** [comment]
No, FileX may cache something that could be updated by USBX.
Recommended sequence:
1. Close media (FileX)
2. Start USBX
3. host may operate the media
4. Stop USBX
5. Open media (FileX)

**Comment by arilink** [comment]
Ok，i will try as you said

**Comment by arilink** [comment]
@xiaocq2001 I take the liberty to ask, are you Chinese?

**Comment by xiaocq2001** [comment]
Yes.

**Comment by arilink** [comment]
Can you leave WeChat? Want to learn more from you

**Comment by derskop** [comment]
hey,sir,can you give me a project about the usbx host msc, recently  i  am studying  how to move it to my project   meeting some diffcult. thanks.

**Comment by xiaocq2001** [comment]
Maybe you can check https://github.com/azure-rtos/samples.
Or you can try to generate USBX host storage demo from STM32 Cube.
